corehq/util/string_utils.py

```diff
@@ -8,7 +9,7 @@ def random_string(n=6):
 
 
 def ensure_unicode(s):
-    if not s or isinstance(s, unicode):
+    if not s or isinstance(s, six.text_type):
```

Comment:
got carried away with Python 3 compatibility last time and didn't notice this `not s` part. That could cause unexpected things to be returned from this function since many things evaluate to false, but this function, according to its name, should always return unicode. What do you think of this instead:
```py
if isinstance(s, six.text_type):
    return s
elif isinstance(s, six.binary_type):
    return s.decode('utf-8')
return six.text_type(s)
```

Reply:
good idea - updated